IPG-Automotive-UK/react-ui
# TD-3238 Create a `LinkWithPreview` component

Closes [TD-3238](https://sce.myjetbrains.com/youtrack/issue/TD-3238)
Contributes [TD-3196](https://sce.myjetbrains.com/youtrack/issue/TD-3196)

## Changes




A brief description of what this pull request solves / introduces.

- Create component `LinkWithPreview` which functions as a normal link, but on hover it will pop-up a component passed into the `content` prop
- Adds tests to check that the popper appears/disappears on user hover/leave after a set time, test to check the contents in the popper are correct

## Dependencies




## UI/UX

[FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=3608-21707&node-type=frame&t=awJoDnSzbvdFuKOZ-0)

Before hovering:
![image](https://github.com/user-attachments/assets/49f8804c-9cda-4572-a3f9-2ada2703d88e)

After hovering:
![image](https://github.com/user-attachments/assets/41a25ad8-e6b6-4cdd-85b3-865be26129a7)

Truncation:
![image](https://github.com/user-attachments/assets/53b3b232-24cd-4e9d-b6a0-f4d31a76112f)





## Testing notes
1. Open storybook from the Github CI/CD processes
2. Navigate to `LinkWithPreview`
3. Make sure you see a link render, on hover it renders an underline.
4. If you hover over the link for more than 1s, make sure that a popper appears
5. On mouse leave, make sure that, after 0.3s the popper closes.
6. Make sure that long link names truncate.



## Author checklist

Before I request a review:



~- [ ] I have populated the deploy-preview with relevant test data.~
